### ISSUE TITLE
Installing Horovod with pip install horovod[mxnet] errors out without building GPU extensions

### ISSUE THREAD

**Issue by davidrpugh** [horovod/horovod#2141]
**Environment:**
1. Framework: MXNet
2. Framework version: 1.6
3. Horovod version: 0.19.5
4. MPI version: 
5. CUDA version: 10.1
6. NCCL version: 2.4
7. Python version: 3.6+
8. Spark / PySpark version:
9. OS and version: CentOS
10. GCC version:

When building Horovod with `HOROVOD_WITH_MXNET=1` and installing Horovod with `pip install horovod[mxnet]`, GPU bindings are not built and an error is thrown.

Issue seems to be that `mxnet_require_list = ['mxnet>=1.4.1']` in the `setup.py` file does not install a GPU enabled builds of MXNet. GPU builds of MXNet have different naming conventions, for example `mxnet-cu101mkl` is the MKL and CUDA 10.1 version of GPU.

Pre-installing the desired version of MXNet and then installing Horovod using `pip install horovod` seems to work as expected.

**Comment by tgaddair** [comment]
Good catch, @davidrpugh.  One option would be to add a separate extra for GPU-enabled MXNet, similar to what we do with TensorFlow.  Though it seems like we would need to do one for potentially every supported CUDA version, with or without MKL.  Is that right?

@eric-haibin-lin any thoughts?

**Comment by davidrpugh** [comment]
For reference here is the list of versions available from PyPI.

```
This package supports Linux, Mac OSX, and Windows platforms. You may also want to check: - mxnet-cu92 with CUDA-9.2 support. - mxnet-cu92mkl with CUDA-9.2 support and MKLDNN support. - mxnet-cu91 with CUDA-9.1 support. - mxnet-cu91mkl with CUDA-9.1 support and MKLDNN support. - mxnet-cu90 with CUDA-9.0 support. - mxnet-cu90mkl with CUDA-9.0 support and MKLDNN support. - mxnet-cu80 with CUDA-8.0 support. - mxnet-cu80mkl with CUDA-8.0 support and MKLDNN support. - mxnet-cu75 with CUDA-7.5 support. - mxnet-cu75mkl with CUDA-7.5 support and MKLDNN support. - mxnet-mkl with MKLDNN support.
```

In the past I have never used extras when installing Horovod. I have always manually installed frameworks(s) and then used combination of environment variables and `pip install horovod` to explicitly control which extensions should be built. As the number of builds available for different frameworks increases it seems the maintenance of extras will become more challenging.

**Comment by eric-haibin-lin** [comment]
sorry for the late reply. @tgaddair yes, that is correct. Although there won't be any mxnet-cuxx-mkl packages for newer mxnet releases, as MKLDNN is now turned on by default. But this option does require separate packages for different cuda versions (e.g. pip install horovod[mxnet-cu100], pip install horovod[mxnet-cu101], etc)
